icu4c/source/test/intltest/itutil.cpp

```diff
@@ -905,3 +907,45 @@ void EnumSetTest::TestEnumSet() {
     assertFalse(WHERE, flags.get(THING2));
     assertFalse(WHERE, flags.get(THING3));
 }
+
+/** UHashTest **/
+#include "uhash.h"
+#include <string_view>
+
+class UHashTest : public IntlTest {
+public:
+    UHashTest() = default;
+    virtual void runIndexedTest(int32_t index, UBool exec, const char*& name, char* par = nullptr) override;
+    void TestStringView();
+};
+
+static IntlTest* createUHashTest() {
+    return new UHashTest();
+}
+
+void UHashTest::runIndexedTest(int32_t index, UBool exec, const char*& name, char*  /*par*/) {
+    TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(TestStringView);
+    TESTCASE_AUTO_END;
+}
+
+void UHashTest::TestStringView() {
+    IcuTestErrorCode status(*this, "TestStringView");
+    LocalUHashtablePointer table(
+        uhash_open(uhash_hashIStringView, uhash_compareIStringView, nullptr, status));
+    if (status.errIfFailureAndReset("uhash_open") ||
+        !assertTrue("uhash_open", table.isValid())) {
+        return;
+    }
+
+    std::string_view key("aaa");
+    std::string_view value("bbb");
+
+    uhash_put(table.getAlias(), &key, &value, status);
+    if (status.errIfFailureAndReset("uhash_put")) return;
+
+    std::string_view lookup = key;
+
+    auto* result = static_cast<std::string_view*>(uhash_get(table.getAlias(), &lookup));
```

Comment:
Please also test with a key that differs in case from one of the ones you put in.

Reply:
Done.